gtests/net/packetdrill/mptcp.c

```diff
@@ -516,8 +517,11 @@ int mptcp_subtype_mp_capable(struct packet *packet_to_modify,
 							mp_state.hash);
 		// If this is done at syn packet time as for inbound, key comparisons fail
 		// due to, I guess, key set too early as it complains key is not 0
-		if(direction == DIRECTION_OUTBOUND)
+		if (optlen == TCPOLEN_MP_CAPABLE_DATA) {
+			/* TODO update SSN counters */
+		} else if (direction == DIRECTION_OUTBOUND) {
```

Comment:
I wonder if this 'else' is correct. It's harmless at the moment, but theoretically the MPC+DATA could be sent as third ack - so, probably we should allocate the first subflow also when `optlen` is 22, unless the first subflow has been already allocated. @matttbe wdyt?

Reply:
@dcaratti 

> I wonder if this 'else' is correct. It's harmless at the moment, 

I guess the `else` is still needed for v0, no?

>  so, probably we should allocate the first subflow also when optlen is 22, unless the first subflow has been already allocated

I am no longer used to this code but it looks like you have to do it at some points. I don't know if you need to wait for the 4th ACK with v1 because I guess it will influence when you can accept extra subflows. I would need to double check if with v1 we can accept new subflows before the 4th ACK.